src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/VCenter/NewAzureRmRecoveryServicesAsrVCenter.cs

```diff
@@ -0,0 +1,126 @@
+﻿// ----------------------------------------------------------------------------------
+// 
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Retrieves Azure Site Recovery vCenter.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.New,
+        "AzureRmRecoveryServicesAsrvCenter",
+        DefaultParameterSetName = ASRParameterSets.Default,
+        SupportsShouldProcess = true)]
+    [Alias("New-ASRvCenter")]
+    [OutputType(typeof(ASRJob))]
+    public class NewAzureRmRecoveryServicesAsrvCenter : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets Fabric of the vCenter.
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public ASRFabric Fabric { get; set; }
```

Comment:
@viverm parameter set for `ResourceId` of Fabric being provided? Please make sure that all cmldets that should support piping have the following three scenarios enabled:

1) By object - involves using an `InputObject` parameter that has `ValueFromPipeline = true`
2) By resource id - involves using a `ResourceId` parameter that has `ValueFromPipelineByPropertyName = true`
3) By command line - involves using a `Name` and `ResourceGroupName` parameter with no ability to accept its value from the pipeline; this is your default parameter set

Reply:
First two scenario fits for ASR. 
Getting the PSObject by Name and ResourceGroupName parameter is not possible in ASR Case we need additional parameter like vault , protection container etc on case to case basis.
Name is mainly server generated Id and hard to remember.